tools/lint/lint.py

```diff
@@ -1005,26 +1027,48 @@ def process_errors(errors):
 
         return (errors[-1][0], path)
 
-    for path in paths[:]:
+    to_check_content = []
+    skip = set()
+
+    for path in paths:
         abs_path = os.path.join(repo_root, path)
         if not os.path.exists(abs_path):
-            paths.remove(path)
+            skip.add(path)
             continue
 
         if any(fnmatch.fnmatch(path, file_match) for file_match in skipped_files):
-            paths.remove(path)
+            skip.add(path)
             continue
 
         errors = check_path(repo_root, path)
         last = process_errors(errors) or last
 
         if not os.path.isdir(abs_path):
-            with io.open(abs_path, 'rb') as test_file:
-                errors = check_file_contents(repo_root, path, test_file)
-                last = process_errors(errors) or last
+            to_check_content.append((repo_root, path))
+
+    paths = [p for p in paths if p not in skip]
```

Comment:
This could be an iterator, right?

Reply:
That doesn't work in the multiprocessing branch, at least with the spawn start method, as iterators aren't pickleable